Common/src/geometry/primal_grid/CPrimalGridFEM.cpp

```diff
@@ -26,67 +26,27 @@
  */
 
 #include "../../../include/geometry/primal_grid/CPrimalGridFEM.hpp"
+#include "../../../include/fem/fem_standard_element.hpp"
 
-CPrimalGridFEM::CPrimalGridFEM(unsigned long val_elemGlobalID, unsigned short val_VTK_Type,
-                               unsigned short val_nPolyGrid, unsigned short val_nPolySol, unsigned short val_nDOFsGrid,
-                               unsigned short val_nDOFsSol, unsigned long val_offDOfsSol, std::istringstream& elem_line)
-    : CPrimalGrid(true, val_nDOFsGrid, nFacesOfElementType(val_VTK_Type)) {
-  /*--- Store the integer data in the member variables of this object. ---*/
-  VTK_Type = val_VTK_Type;
-  nFaces = nFacesOfElementType(VTK_Type);
-
-  nPolyGrid = val_nPolyGrid;
-  nPolySol = val_nPolySol;
-  nDOFsGrid = val_nDOFsGrid;
-  nDOFsSol = val_nDOFsSol;
-
-  elemIDGlobal = val_elemGlobalID;
-  offsetDOFsSolGlobal = val_offDOfsSol;
-
-  /*--- Read face structure of the element from elem_line. ---*/
-
-  for (unsigned short i = 0; i < nDOFsGrid; i++) elem_line >> Nodes[i];
-
-  /*--- If a linear element is used, the node numbering for non-simplices
-        must be adapted. The reason is that compatability with the original
-        SU2 format is maintained for linear elements, but for the FEM solver
-        the nodes of the elements are stored row-wise.                       ---*/
-  if (nPolyGrid == 1) {
-    switch (VTK_Type) {
-      case QUADRILATERAL:
-        std::swap(Nodes[2], Nodes[3]);
-        break;
+CPrimalGridFEM::CPrimalGridFEM(const unsigned long* dataElem, unsigned long& offsetSolDOFs)
+    : CPrimalGrid(true, dataElem[3], nFacesOfElementType(dataElem[0])) {
+  /*--- Store the meta data for this element. ---*/
+  VTK_Type = (unsigned short)dataElem[0];
+  nPolyGrid = (unsigned short)dataElem[1];
+  nPolySol = (unsigned short)dataElem[2];
+  nDOFsGrid = (unsigned short)dataElem[3];
```

Comment:
Try to replace this type of cast with static_cast<>() please. Refactoring looks good :+1:

Reply:
Replace the C-style casts by static_cast.